File: song.py
def main():
    sing()


def sing():
    #Make loop not recursive
    while True:
        print("This is the song that doesn't end.")
        print("Yes, it goes on and on my friend.")
        print("Some people started singing it not knowing what it was,")
        print("And they'll continue singing it forever just because...")
        sing()


if __name__ == "__main__":
    main()

File: song.c
#include <cs50.h>
#include <stdio.h>

void recursion():
{
    printf("This is the song that doesn't end.");
    printf("Yes, it goes on and on my friend.");
    printf("Some people started singing it not knowing what it was,");
    printf("And they'll continue singing it forever just because...");
    recursion();
}

int main (void)
{
    recursion();
}